### PR TITLE
sc-2693 SelectPeer tests

### PR DESCRIPTION
**Pull request by pdeziel**
This adds some tests for the SelectPeer method in the replica service. `TestSelectZeroOrOnePeer` is a table test which tests the edge cases. `TestSelectPeerAll` is a more comprehensive test which makes sure that we are selecting uniformly random peers and that we are never selecting the local replica for gossip.

### CODE REVIEW

File: pkg/trtl/replica/sync_test.go [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+	require.NoError(t, err)
+	require.NoError(t, json.Unmarshal(data, &fixtures), "could not unmarshal peers fixtures")
+	require.Len(t, fixtures, 4, "unexpected number of peers fixtures")
+	return fixtures
+}
+
+func createDB(t *testing.T, fixtures []*peers.Peer) *honu.DB {
+	tmp, err := ioutil.TempDir("testdata", "*-db")
+	require.NoError(t, err)
+	t.Cleanup(func() { os.RemoveAll(tmp) })
```
Comment (bbengfort) [inline pkg/trtl/replica/sync_test.go:96]: Nice use of cleanup!

File: pkg/trtl/replica/sync_test.go [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+	db, err := honu.Open("leveldb:///" + tmp)
+	require.NoError(t, err)
+
+	// Populate the database with the peers
+	for _, p := range fixtures {
+		p.Created = time.Now().Format(time.RFC3339)
+		p.Modified = p.Created
```
Comment (bbengfort) [inline pkg/trtl/replica/sync_test.go:104]: Nice! Thank you for populating the entire record!

File: pkg/trtl/replica/sync_test.go [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+			Region:         self.Region,
+			GossipInterval: 10 * time.Minute,
+			GossipSigma:    1500 * time.Millisecond,
+		},
+		MTLS: config.MTLSConfig{
+			Insecure: true,
+		},
+	}
+
+	replica, err := replica.New(conf, db, nil)
```
Comment (bbengfort) [inline pkg/trtl/replica/sync_test.go:130]: At least the replica is slightly easier to mock since we're passing the conf and db directly into it!

File: pkg/trtl/replica/sync_test.go [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+	// Run enough times so that all peers have a chance to be selected.
+	counter := make(map[uint64]uint8)
+	for i := 0; i < 100; i++ {
+		peer := replica.SelectPeer()
+		require.NotNil(t, peer, "should have selected a peer")
+		counter[peer.Id]++
+	}
+
+	// All peers except the local replica should be selected.
+	require.Len(t, counter, len(peers)-1, "did not select all peers")
```
Comment (bbengfort) [inline pkg/trtl/replica/sync_test.go:72]: Good use of the fixtures struct in testing - I'm glad this test is setup so we can easily change the peers fixtures!

**Review by bbengfort — APPROVED** [review]
Excellent tests - I'm glad we were also testing the edge cases.
